parse/src/lib.rs

```diff
@@ -65,10 +66,52 @@ struct SolTxMetaInfo {
 struct TransactionMeta {
     neon_transaction: Transaction,
     sol_ix_idx: usize,
+    sol_ix_inner_idx: Option<u64>,
+    neon_ix_code: u8, // TODO: Check,
+
     is_cancelled: bool,
     is_completed: bool,
 }
 
+fn parse_cu_info(tx: &VersionedTransaction, loaded: &LoadedAddresses) -> SolTxCuInfo {
+    let pubkeys = AccountKeys::new(tx.message.static_account_keys(), Some(loaded));
+    let cu_idx = pubkeys.iter().position(|x| *x == compute_budget::id());
+
+    let mut sol_tx_cu_info = SolTxCuInfo::default();
+
+    for ix in tx.message.instructions() {
+        tracing::info!("instruction {:?}", ix); // TODO: downgrade to debug
+        if let Some(cu_idx) = cu_idx {
+            if ix.program_id_index as usize == cu_idx {
+                tracing::info!("we are having the prog"); // TODO: remove
+                match ix.data.first() {
+                    Some(0) => {
+                        sol_tx_cu_info.heap_size = u32::from_le_bytes(
+                            ix.data[1..5].try_into().expect("Failed to parse heap size"), // todo: error
+                        );
+                    }
+                    Some(1) => {
+                        sol_tx_cu_info.cu_limit = u32::from_le_bytes(
+                            ix.data[1..5]
+                                .try_into()
+                                .expect("Failed to parse compute unit limit"), // todo: error
+                        );
+                    }
+                    Some(2) => {
+                        sol_tx_cu_info.cu_price = u32::from_le_bytes(
+                            ix.data[1..5]
+                                .try_into()
+                                .expect("Failed to parse compute unit price"),
+                        ); // todo: error
+                    }
+                    _ => tracing::warn!("unknown Compute Budget instruction"),
```

Comment:
This log lacks information about instruction, unexpected tag and related transaction